app/src/main/java/chat/rocket/android/main/ui/MainActivity.kt

```diff
@@ -279,4 +284,79 @@ class MainActivity : AppCompatActivity(), MainView, HasActivityInjector,
         progressDialog?.dismiss()
         progressDialog = null
     }
-}
+
+    fun changeLanguage() {
+        val languages = resources.getStringArray(R.array.languages)
+
+        val builder = AlertDialog.Builder(this)
+        builder.setTitle(R.string.title_choose_language)
+        builder.setSingleChoiceItems(languages, -1) { dialog, which ->
+            when(which){
+                0->{
+                    setLocale("en")
+                    recreate()
+                }
+                1->{
+                    setLocale("hi")
+                    recreate()
+                }
+                2->{
+                    setLocale("ja")
+                    recreate()
+                }
+                3->{
+                    setLocale("ru")
+                    recreate()
+                }
+                4->{
+                    setLocale("it")
+                    recreate()
+                }
+                5->{
+                    setLocaleWithRegion("pt","BR")
+                    recreate()
+                }
+                6->{
+                    setLocaleWithRegion("pt", "PT")
+                    recreate()
+                }
+                7->{
+                    setLocale("zh")
+                    recreate()
+                }
+            }
+            dialog.dismiss()
+        }
+        builder.create().show()
+    }
+
+    private fun setLocale(lang: String) {
```

Comment:
`setLocale`, `setLocaleWithRegion` and `loadLocale` = Too much responsibility to  `MainActivity.kt`

Take a look on how we are doing basically this kind of thing on the code (`SharedPrefsConnectingServerRepository`, `SharedPrefsAnalyticsTrackingRepository` and others...)

Reply:
Yes, the MainActivity.kt is having too much responsibility, I will look into the implementation of SharedPrefsConnectingServerRepository and push the changes soon 👍